syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.graphsearch";
option java_outer_classname = "GraphSearchProto";
option objc_class_prefix = "GS";

package graphsearch;

service GraphSearchService {
  rpc GraphSearch (GraphSearchRequest) returns (GraphSearchReply) {}
  rpc TrustedValidation (GraphSearchRequest) returns (TrustedValidationReply) {}
}

message GraphSearchRequest {
    string txid = 1;
    repeated string exclude_txids = 2;
}

message GraphSearchReply {
    repeated bytes txdata = 1;
}

message TrustedValidationReply {
    bool valid = 1;
}
